glean-core/src/lib.rs

```diff
@@ -65,6 +65,8 @@ pub struct Configuration {
     pub application_id: String,
     /// The maximum number of events to store before sending a ping containing events.
     pub max_events: Option<usize>,
+    /// Whether Glean should defer ping collection.
+    pub defer_collection: Option<bool>,
```

Comment:
Why is this an option? What do the three states it could be in mean (`Some(true), Some(false), None`)?

Reply:
Actually, do we need to store that value at all in the Glean struct?
The database already knows if it has the ping metrics store, no need to keep this flag around at all

Reply:
Hm, now that you say it, this doesn't really need to be an option.

> The database already knows if it has the ping metrics store, no need to keep this flag around at all

I get this part, can you explain it a little bit further?

Reply:
Ah sorry. Misread the code, ignore that part, I thought this was the glean object.
It's the configuration object, so of course we need it